packages/hardhat-core/src/internal/hardhat-network/provider/utils/makeFakeSignature.ts

```diff
@@ -0,0 +1,23 @@
+import crypto from "crypto";
+import util from "util";
+
+// Produces a signature with r and s values taken from a hash of the inputs.
+export function makeFakeSignature(...inputs: any[]): {
+  v: number;
+  r: number;
+  s: number;
+} {
+  const hash = crypto.createHash("md5");
+
+  for (const input of inputs) {
+    hash.update(Buffer.from(`${util.inspect(input)}`));
```

Comment:
This part concerns me a bit, because the `util.inspect` function could return a different value after, say, installing a library that patches the inspect function of some type. I don't know how likely this is to happen, but I think we should avoid it just in case.

The alternative that comes to mind is to just use the sender address and the nonce of the tx here, since those should be unique enough.

Reply:
> The alternative that comes to mind is to just use the sender address and the nonce of the tx here, since those should be unique enough.

This can be problematic when you want to replace a pending tx.


The problem here is that `any[]`. I think we can make this type-safe and use appropriate serializations.

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/3175/commits/a4118e0f410b44697d5da3a3c8fcbff0acbe8a43

But did I address it entirely? I made it type-safe, but I wasn't sure about the "and use appropriate serializations" part.